provisioning/ubuntu-provision.sh

```diff
@@ -198,6 +198,23 @@ function install_docker() {
   usermod -aG docker "${username}"
 }
 
+## Ensure Golang is installed
+function install_golang(){
+  local temp_archive="/tmp/go${GOLANG_VERSION}.tgz"
+  local installation_dir="/usr/local/"
+  local download_url="https://go.dev/dl/go${GOLANG_VERSION}.linux-${ARCHITECTURE}.tar.gz"
+  curl --fail --silent --show-error --location --output "${temp_archive}" "${download_url}"
+  tar --extract --gunzip --file="${temp_archive}" --directory="${installation_dir}"
+  # Symlink to a folder in the PATH
+  ln -s "${installation_dir}go/bin/go" /usr/local/bin/go
+}
+
+## Ensure GolangCIlint is installed
+function install_golangcilint(){
+  # binary will be in /usr/local/bin
+  curl -sSfL "https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh" | sh -s -- -b /usr/local/bin "v${GOLANGCILINT_VERSION}"
```

Comment:
Not sure about the execution of this external script.

Reply:
I would extract the main commands/logic from this remote script to our own script here. (download from a versioned URL + checksum + tar)